### PR TITLE
Tracks Channel ID

### PR DESCRIPTION
**Pull request by mtuchkov**
Motivation:
During investigation it is impossible to distinguish events sent by one client
from event sent by another client

Modifications:
Every event traces channel ID as a new column in ETW events

Results:
All events that happen during activity of one client have unique distinguisher
-  the channel ID, that can be used for the purpose described above.
We also trace event that matches channel ID with device ID so that users could
easily filter events made by a specific client.

### CODE REVIEW

File: src/ProtocolGateway.Core/Mqtt/MqttAdapter.cs [hunk 1]
```diff
@@ -105,12 +105,12 @@ public override void ChannelActive(IChannelHandlerContext context)
...
 
             TimeSpan? ackTimeout = this.settings.DeviceReceiveAckCanTimeout ? this.settings.DeviceReceiveAckTimeout : (TimeSpan?)null;
-            this.publishPubAckProcessor = new RequestAckPairProcessor<AckPendingMessageState, PublishPacket>(this.AcknowledgePublishAsync, this.RetransmitNextPublish, ackTimeout);
-            this.publishPubAckProcessor.Completion.OnFault(ShutdownOnPubAckFaultAction);
-            this.publishPubRecProcessor = new RequestAckPairProcessor<AckPendingMessageState, PublishPacket>(this.AcknowledgePublishReceiveAsync, this.RetransmitNextPublish, ackTimeout);
-            this.publishPubRecProcessor.Completion.OnFault(ShutdownOnPubRecFaultAction);
-            this.pubRelPubCompProcessor = new RequestAckPairProcessor<CompletionPendingMessageState, PubRelPacket>(this.AcknowledgePublishCompleteAsync, this.RetransmitNextPublishRelease, ackTimeout);
-            this.pubRelPubCompProcessor.Completion.OnFault(ShutdownOnPubCompFaultAction);
+            this.publishPubAckProcessor = new RequestAckPairProcessor<AckPendingMessageState, PublishPacket>(this.AcknowledgePublishAsync, this.RetransmitNextPublish, ackTimeout, context.Channel.Id.ToString());
```
Comment (nayato) [inline src/ProtocolGateway.Core/Mqtt/MqttAdapter.cs]: >context.Channel.Id.ToString() [](start = 179, length = 29)

DRY: property

File: src/ProtocolGateway.Core/Mqtt/MqttAdapter.cs [hunk 2]
```diff
@@ -788,6 +789,9 @@ bool TryMatchSubscription(string topicName, DateTime messageTime, out QualityOfS
                 this.stateFlags = StateFlags.ProcessingConnect;
                 this.identity = await this.authProvider.GetAsync(packet.ClientId,
                     packet.Username, packet.Password, context.Channel.RemoteAddress);
+
+                CommonEventSource.Log.Info("Client ID" + this.identity, string.Empty, context.Channel.Id.ToString());
```
Comment (nayato) [inline src/ProtocolGateway.Core/Mqtt/MqttAdapter.cs]: >Client ID [](start = 44, length = 9)

Authenticated or not?

**Comment by nayato** [comment]
:shipit:

File: src/ProtocolGateway.Core/Mqtt/MqttAdapter.cs [hunk 3]
```diff
@@ -800,11 +804,13 @@ bool TryMatchSubscription(string topicName, DateTime messageTime, out QualityOfS
                     return;
                 }
 
+                CommonEventSource.Log.Info("ClientAuthenticated", this.identity.ToString(), this.ChannelId);
```
Comment (robichaud) [inline src/ProtocolGateway.Core/Mqtt/MqttAdapter.cs]: ``` IDeviceIndentity ``` implementations do not override the ```ToString() ``` method. You will only log the fully qualified class name here. Either implement `` ToString() `` or change this to ``` this.identity.Id ```

Comment (robichaud) [inline src/ProtocolGateway.Core/Mqtt/MqttAdapter.cs]: Looks like String Interpolation is used all over place with the IDeviceIdentity object.

Comment (mtuchkov) [inline src/ProtocolGateway.Core/Mqtt/MqttAdapter.cs]: Done. Please take a look

Comment (nayato) [inline src/ProtocolGateway.Core/Mqtt/MqttAdapter.cs]: LGTM now

**Comment by nayato** [comment]
looks good now

**Pull request merged**